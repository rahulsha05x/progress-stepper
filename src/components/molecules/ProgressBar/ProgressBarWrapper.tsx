import React from 'react';

export interface StepData {
  text?: string | object;
  status?: string;
  id: number;
  className: string;
  progress?: number;
}
interface Props {
  steps: StepData[];
  activeStep: number;
  className: string;
  orientation?: string;
  color?: string;
  content?: any;
}

export const getStatus = (activeStep: number, index: number) => {
  let stepClass = 'inactive';
  const completed = activeStep > index + 1;
  if (completed) {
    stepClass = 'is-completed';
  }
  if (activeStep === index + 1) {
    stepClass = 'is-active';
  }
  return stepClass;
};
const ProgressBarWrapper: React.FC<any> = ({
  steps,
  activeStep,
  align = 'center',
  orientation = 'horizontal',
  color,
  content,
  className = '',
  children,
}) => {
  let currentItemStatus: string;
  const renderChildren = () => {
    return React.Children.map(children, (child, index) => {
      currentItemStatus = getStatus(activeStep, index);
      return React.cloneElement(child, {
        currentitemstatus: currentItemStatus,
        itemindex: index,
      });
    });
  };
  const renderChild = renderChildren();
  return (
    <ul
      data-testid="progressBar-wrapper"
      className={`progress-tracker --${orientation} ${className}`}
    >
      {renderChild}
    </ul>
  );
};

export default ProgressBarWrapper;
