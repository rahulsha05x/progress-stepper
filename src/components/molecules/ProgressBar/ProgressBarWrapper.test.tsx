import React from 'react';
import { render, screen } from '@testing-library/react';
import ProgressBarWrapper from './ProgressBarWrapper';

describe('ProgressBarWrapper', () => {
  test('Renders ProgressBar wrapper', () => {
    const children = <li>Child example</li>;
    render(
      <ProgressBarWrapper activeStep={1} orientation="vertical">
        {children}
      </ProgressBarWrapper>
    );
    expect(screen.getByTestId('progressBar-wrapper')).toMatchSnapshot();
    expect(screen.getByTestId('progressBar-wrapper')).toHaveClass(
      'progress-tracker',
      '--vertical'
    );
    expect(screen.getByTestId('progressBar-wrapper')).toContainHTML(
      '<li currentitemstatus="is-active" itemindex="0">Child example</li>'
    );
  });
  test('Renders ProgressBar wrapper', () => {
    const children = [<li>Child example</li>, <li>Child example2</li>];
    render(
      <ProgressBarWrapper activeStep={2}>
        {children[0]}
        {children[1]}
      </ProgressBarWrapper>
    );
    expect(screen.getByTestId('progressBar-wrapper')).toHaveClass(
      'progress-tracker',
      '--horizontal'
    );

    expect(screen.getByTestId('progressBar-wrapper')).toContainHTML(
      '<li currentitemstatus="is-completed" itemindex="0">Child example</li><li currentitemstatus="is-active" itemindex="1">Child example2</li>'
    );
  });
});
