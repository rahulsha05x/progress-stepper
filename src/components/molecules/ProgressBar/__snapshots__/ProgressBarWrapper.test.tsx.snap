// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBarWrapper Renders ProgressBar wrapper 1`] = `
<ul
  class="progress-tracker --vertical "
  data-testid="progressBar-wrapper"
>
  <li
    currentitemstatus="is-active"
    itemindex="0"
  >
    Child example
  </li>
</ul>
`;
